File: app/src/main/java/wjy/yo/ereader/entityvo/book/ChapDetail.java
package wjy.yo.ereader.entityvo.book;

import android.arch.persistence.room.Relation;

import java.util.List;
import java.util.Objects;

import wjy.yo.ereader.entity.book.Chap;
import wjy.yo.ereader.entity.book.Para;

public class ChapDetail extends Chap {

    @Relation(parentColumn = "id", entityColumn = "chapId")
    private List<Para> paras;

    public List<Para> getParas() {
        return paras;
    }

    public void setParas(List<Para> paras) {
        this.paras = paras;
    }

    @Override
    public boolean equals(Object o) {
        if (o == null) {
            return false;
        }
        if (!super.equals(o)) {
            return false;
        }
        ChapDetail cd = (ChapDetail) o;
        return Objects.equals(paras, cd.paras);
    }

}

File: app/src/main/java/wjy/yo/ereader/entityvo/book/ChapDetail.kt
package wjy.yo.ereader.entityvo.book

import android.arch.persistence.room.Relation
import java.util.Objects

import wjy.yo.ereader.entity.book.Chap
import wjy.yo.ereader.entity.book.Para

class ChapDetail : Chap() {

    @Relation(parentColumn = "id", entityColumn = "chapId")
    var paras: List<Para>? = null

    override fun equals(other: Any?): Boolean {
        if (other == null) {
            return false
        }
        if (!super.equals(other)) {
            return false
        }
        val cd = other as ChapDetail?
        return paras == cd!!.paras
    }

}
